### PR TITLE
Feature/add test pypi publish add spectree link

### PR DESCRIPTION
**Pull request by cgearing**
This adds a link to Spectree in the docs, and adds an Action to push the library to Test PyPi on release.

### CODE REVIEW

File: .github/workflows/publish-test.yaml [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+      with:
+        python-version: '3.x'
+    - name: Install dependencies
+      run: |
+        python -m pip install --upgrade pip
+        pip install setuptools wheel twine
+    - name: Build and publish
+      env:
+        TWINE_USERNAME: ${{ secrets.PYPI_USERNAME }}
+        TWINE_PASSWORD: ${{ secrets.TEST_PYPI_KEY }}
```
Comment (simonhayward) [inline .github/workflows/publish-test.yaml:28]: Oh this is going to be published..nice

Comment (cgearing) [reply .github/workflows/publish-test.yaml:28]: Yeah just for testing - I'll do a test push!

File: .github/workflows/publish-test.yaml [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+
+    steps:
+    - uses: actions/checkout@v2
+    - name: Set up Python
+      uses: actions/setup-python@v2
+      with:
+        python-version: '3.x'
+    - name: Install dependencies
+      run: |
+        python -m pip install --upgrade pip
```
Comment (robyoung) [inline .github/workflows/publish-test.yaml:23]: Do you need to install pip? That should come by default with python 3

File: .github/workflows/publish-test.yaml (hunk 1 above)
Comment (robyoung) [inline .github/workflows/publish-test.yaml:28]: Are you uploading this to the test pypi instance? If so, is there something that tells twine to do that?

**Review by robyoung — COMMENTED** [review]
Looks great. I have a couple of comments but nothing drastic.

Have you seen that PyPA have a github action for publishing packages? https://github.com/marketplace/actions/pypi-publish It may not be ideal as it leaves building up to us (which we'd probably do with twine) however it does have some interesting stuff about using API tokens rather than username and password.

**Comment by robyoung** [comment]
We should probably rename the existing workflow as it's called "Python package" but it doesn't do any packaging :confused:

File: .github/workflows/publish-test.yaml (hunk 2 above)
Comment (cgearing) [reply .github/workflows/publish-test.yaml:23]: I actually took this from the github action template it suggested and edited for different secrets, but I have also just seen this: https://packaging.python.org/guides/publishing-package-distribution-releases-using-github-actions-ci-cd-workflows/ so I might redo it following this?

Comment (cgearing) [reply .github/workflows/publish-test.yaml:23]: I think they use the same action

Comment (robyoung) [reply .github/workflows/publish-test.yaml:23]: oh nice!

Comment (cgearing) [reply .github/workflows/publish-test.yaml]: Absolutely useless documentation.
`In this example, we’ll use pep517 package, assuming that your project has a pyproject.toml properly set up (see PEP 517/PEP 518).`

and then they don't give you any help on what a proper `pyproject.toml` is or looks like.

**Review by robyoung — APPROVED** [review]

**Pull request merged**